src/plugins/discover/public/application/apps/main/discover_main_app.tsx

```diff
@@ -12,45 +12,41 @@ import { setBreadcrumbsTitle } from '../../helpers/breadcrumbs';
 import { addHelpMenuToAppChrome } from '../../components/help_menu/help_menu_util';
 import { useDiscoverState } from './services/use_discover_state';
 import { useUrl } from './services/use_url';
-import { IndexPattern, IndexPatternAttributes, SavedObject } from '../../../../../data/common';
+import { IndexPatternAttributes, SavedObject } from '../../../../../data/common';
 import { DiscoverServices } from '../../../build_services';
 import { SavedSearch } from '../../../saved_searches';
 
 const DiscoverLayoutMemoized = React.memo(DiscoverLayout);
 
 export interface DiscoverMainProps {
   /**
-   * Current IndexPattern
+   * Instance of browser history
    */
-  indexPattern: IndexPattern;
-
-  opts: {
-    /**
-     * Use angular router for navigation
-     */
-    navigateTo: () => void;
-    /**
-     * Instance of browser history
-     */
-    history: History;
-    /**
-     * List of available index patterns
-     */
-    indexPatternList: Array<SavedObject<IndexPatternAttributes>>;
-    /**
-     * Kibana core services used by discover
-     */
-    services: DiscoverServices;
-    /**
-     * Current instance of SavedSearch
-     */
-    savedSearch: SavedSearch;
-  };
+  history: History;
+  /**
+   * List of available index patterns
+   */
+  indexPatternList: Array<SavedObject<IndexPatternAttributes>>;
+  /**
+   * Kibana core services used by discover
+   */
+  services: DiscoverServices;
+  /**
+   * Current instance of SavedSearch
+   */
+  savedSearch: SavedSearch;
 }
 
 export function DiscoverMainApp(props: DiscoverMainProps) {
-  const { services, history, navigateTo, indexPatternList } = props.opts;
+  const { services, history, indexPatternList } = props;
   const { chrome, docLinks, uiSettings: config, data } = services;
+  const navigateTo = useCallback(
+    (path: string) => {
+      history.push(path);
```

Comment:
While comparing routing navigation found one bug not related to this changes. After hiding the chart and then clicking to `new`, we can't enable histogram any more.  


https://user-images.githubusercontent.com/39378793/130099640-dc24f04d-29a7-49f3-99c3-b623c0236721.mp4

Reply:
Good catch, managed to reproduce this on master. Will open an issue for this.

Reply:
https://github.com/elastic/kibana/issues/110061